src/Symfony/Component/Form/Extension/Core/DataTransformer/DateTimeToHTML5DateTimeLocalTransformer.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Form\Extension\Core\DataTransformer;
+
+use Symfony\Component\Form\Exception\TransformationFailedException;
+
+/**
+ * @author Franz Wilding <franz.wilding@me.com>
+ * @author Bernhard Schussek <bschussek@gmail.com>
+ */
+class DateTimeToHTML5DateTimeLocalTransformer extends BaseDateTimeTransformer
+{
+    /**
+     * Transforms a normalized date into a localized date without trailing timezone.
+     *
+     * According to the HTML standard, the input string of a datetime-local
+     * input is a rfc3339 datetime string without timezone suffix.
+     *
+     * @param \DateTime|\DateTimeInterface $dateTime A DateTime object
+     *
+     * @return string The formatted date
+     *
+     * @throws TransformationFailedException If the given value is not an
+     *                                       instance of \DateTime or \DateTimeInterface
+     */
+    public function transform($dateTime)
+    {
+        if (null === $dateTime) {
+            return '';
+        }
+
+        if (!$dateTime instanceof \DateTime && !$dateTime instanceof \DateTimeInterface) {
+            throw new TransformationFailedException('Expected a \DateTime or \DateTimeInterface.');
+        }
+
+        if ($this->inputTimezone !== $this->outputTimezone) {
+            if (!$dateTime instanceof \DateTimeImmutable) {
+                $dateTime = clone $dateTime;
+            }
+
+            $dateTime = $dateTime->setTimezone(new \DateTimeZone($this->outputTimezone));
+        }
+
+        return preg_replace('/\+00:00$/', '', $dateTime->format('c'));
+    }
+
+    /**
+     * Transforms a formatted string following RFC 3339 into a normalized date.
+     *
+     * @param string $rfc3339 Formatted string
```

Comment:
The reference to RFC3339 here and anywhere else in this class is now not true anymore, is it?